src/notation/internal/notationinteraction.cpp

```diff
@@ -3504,3 +3505,35 @@ void NotationInteraction::toggleUnderline()
 {
     toggleFontStyle(Ms::FontStyle::Underline);
 }
+
+void NotationInteraction::loadStyle(const mu::io::path& stylePath)
+{
+    QFile f(stylePath.toQString());
+    if (!f.open(QIODevice::ReadOnly) || !Ms::MStyle::isValid(&f)) {
+        interactive()->error(trc("notation", "The style file could not be loaded."),
+                             f.errorString().toStdString(), { IInteractive::Button::Ok },
+                             IInteractive::Button::Ok, IInteractive::Option::WithIcon);
+        return;
+    }
+
+    startEdit();
+    if (!score()->loadStyle(stylePath.toQString()) && interactive()->warning(
+            trc("notation",
+                "Since this style file is from a different version of MuseScore, your score is not guaranteed to display correctly."),
```

Comment:
It looks like we need to return some error code from loadStyle (not just bool). We can't guarantee that the "old version" is the reason for the failure

Reply:
That's why I added the isValid check. MU3 didn't even have that!  Agree that returning "false" to mean version mismatch isn't ideal but  I'm not sure it's worth investing too much time improving that particular code right now.